### PR TITLE
Cleanup, add type annotations and improve docstrings

### PR DESCRIPTION
**Pull request by JamieJQuinn**
This PR cleans up some unused flags in `load_source`, adds type annotations where appropriate and broadens the docstrings to act as potential sources for auto-generation of documentation with Sphinx in the near future. The docstrings use [google's style](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) for maximum readability and compatibility with Sphinx.

### CODE REVIEW

File: pycascadia/remove_restore.py [hunk 1]
```diff
@@ -14,8 +15,52 @@
...
+
+    This works by creating a grid containing 1 where there is data in the input grid and a 0 where there is none, then applying a filter to this grid to smooth the boundary. This can be directly multiplied by the difference grid to smooth its edges.
+
+    Args:
+        diff_grid: difference grid to calculate smoothing filter from
+        nodata_val: value in grid representing a lack of data
+        window_width: width of smoothing region at data boundary
+
+    Returns:
+        grid which should be multiplied by input grid in order to smooth
```
Comment (alessandrofelder) [inline pycascadia/remove_restore.py:32]: this docs are very clear 🚀

File: scripts/close_boundary.py [hunk 2]
```diff
@@ -5,41 +5,62 @@
...
-    parser = argparse.ArgumentParser(description='Add halo surrounding given netcdf file')
-    parser.add_argument('--input', required=True, help='input file')
-    parser.add_argument('--output', required=True, help='output file')
-    parser.add_argument('--value', type=float, default=0.0, help='value to replace boundary with')
-    parser.add_argument('--offset', action='store_true', default=False, help='use one index from true boundary as boundary')
-    parser.add_argument('--plot', action='store_true', default=False, help='plot final grid before saving')
-    parser.add_argument('--region', required=False, metavar=('xmin', 'xmax', 'ymin', 'ymax'), nargs=4, type=float,
-                        help='output region. Defaults to the extent of the input grid.')
+    parser = argparse.ArgumentParser(
+        description="Add halo surrounding given netcdf file"
```
Comment (alessandrofelder) [inline scripts/close_boundary.py:19]:
```suggestion
        description="Add piecewise halo surrounding given netcdf file"
```
Just a suggestion, not sure it's the best. I wonder whether just "halo" implies continuity all around the grid.

**Review by alessandrofelder — CHANGES_REQUESTED** [review]
Great work, @JamieJQuinn!
I've made some minor comments for you to consider.

Below are some style points that I may be wrong on (and that probably `black` can take care of?) - we can also just be happy that `black` is happy 😆 

Generally, from looking at the Google style guide, I think that
- the return values and the argument descriptions should be capitalised,
- doc lines should end with a period

and this is not consistently the case,  e.g.
```comment
    Args:
        fname: filename of input grid             # inconsistent
        region: Optional region to crop to.        # correct
 ```
 And similarly for other parts of the docstring.

Very nitpickily, we might want to spell [`netCDF`](https://www.unidata.ucar.edu/software/netcdf/) and [`GeoTiff`](https://earthdata.nasa.gov/esdis/eso/standards-and-references/geotiff) like so in our docstrings, to be consistent with the spelling of their webpages (although `netCDF`/`NetCDF` seem to be used interchangeably to some extent by them?)

File: pycascadia/grid.py [hunk 3]
```diff
@@ -1,57 +1,106 @@
...
+        os.system(
+            f"gmt grdsample {in_fname} -G{out_fname} -R{region_to_str(self.region)} -I{spacing} -V"
+        )
         self.load(out_fname)
         os.remove(in_fname)
         os.remove(out_fname)
 
-    def as_xyz(self):
-        """Returns pandas dataframe representation"""
+    def as_xyz(self) -> pandas.DataFrame:
```
[the remark this answers is not in the document]
Comment (JamieJQuinn) [reply pycascadia/grid.py:82]: My thinking is that it's conceptually a collection of xyz data points, as opposed to a grid. We've broadly kept the terms grid -> xarray.DataArray and xyz -> pandas.DataFrame so I don't think it's too important to make the distinction that it's a dataframe. Perhaps in the future if another xyz representation ever gets added, the distinction would be more important?

[the remark this answers is not in the document]
Comment (alessandrofelder) [reply pycascadia/grid.py:82]: OK, right, I am happy with that.

**Review by alessandrofelder — APPROVED** [review]
Awesome 💯
I'll leave the merging honours to you, @JamieJQuinn :)

**Pull request merged**